sanic/handlers/directory.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+from datetime import datetime
+from operator import itemgetter
+from pathlib import Path
+from stat import S_ISDIR
+from typing import Dict, Iterable, Sequence, Union, cast
+
+from sanic.exceptions import NotFound
+from sanic.pages.autoindex import AutoIndex, FileInfo
+from sanic.request import Request
+from sanic.response import file, html, redirect
+
+
+class DirectoryHandler:
+    def __init__(self, debug: bool) -> None:
+        self.debug = debug
```

Comment:
Why, anyway, is debug argument copied all around Sanic, instead of always accessing it from config?

Passing around variables should generally be avoided, so this (wide) practice of Sanic has always been a mystery to me. This PR seems to add about four more such variables (the two we've discussed elsewhere, debug, and the `directory_handler` argument). There's gotta be better ways to do this but while a bigger refactoring is not a nice idea, I would to a highest degree resist adding even more.

Reply:
FWIW, Java coders apparently invented Singleton Pattern to covertly use global variables (shh, don't tell anyone), and webfront devs are all over a shiny new thing "state management" now because they too were passing (too many) things by arguments from component to another just like Sanic is now, and a state manager conveniently provides the whole application access to them without any arguments being passed through long chains of components.

I am pretty sure that we could very well be using a module-level storage somewhere to get access to app or other relevant objects from which the config, routes and other such things can be extracted.

Reply:
That is not possible because we _expressly_ allow for multiple applications. Therefore, while we have `Sanic.get_app()` that allows app developers this convenience, it is not something that Sanic internals can make use of.

Reply:
Okay, in that case we should be passing the app (or request from where one can get the app) around more often than a list of too many kwargs.

I didn't properly look at the whole architecture yet, but is there a chance of removing this entire DirectoryHandler class? To me the current implementation looks like it only needs the handle function whose code should be part of the static handler, and then returning `response.file_browser` in some cases.

Reply:
I.e. probably what its handle function currently does (returns index file or returns browser, a few lines of code) should go into `sanic.mixins.routes._static_request_handler` instead.

Reply:
Well, now that it has been removed from `file(...)` it does change the architecture a bit.